File: Go/Easy/Insertion Sort/solution_1.go
package main

func InsertionSort(array []int) []int {

	for i := 1; i < len(array); i++ {

		j := i

		for j >= 0 && array[j] < array[j-1] {
			swap(j, j-1, array)
			j -= 1

		}
	}

	return array
}

func swap(index int, indexTwo int, array []int) {
	array[index], array[indexTwo] = array[indexTwo], array[index]
}

File: Python/Easy/Insertion Sort/solution_1.py
def insertionSort(array):

    for i in range(1, len(array)):
        j = i

        while j > 0 and array[j] < array[j-1]:
            swap(j, j - 1, array)
            j -= 1
    return array


def swap(index, index_two, array):
    array[index] , array[index_two] = array[index_two] , array[index]